source/features/selection-in-new-tab.tsx

```diff
@@ -16,14 +15,14 @@ function openInNewTab({key, target}: KeyboardEvent): void {
 	}
 }
 
-function init(): void {
-	document.addEventListener('keypress', openInNewTab);
+function init(signal: AbortSignal): void {
+	document.addEventListener('keypress', openInNewTab, {signal});
 }
 
 void features.add(import.meta.url, {
 	shortcuts: {
 		'shift o': 'Open selection in new tab',
 	},
 	awaitDomReady: false,
-	init: onetime(init),
+	init,
```

Comment:
This however is one of those situations where listeners will just pile up in Safari, and that's a big bug. :(

Potential solution: our own addEventListener wrapper:

```JavaScript 
function onEvent(element, name, signal, handler, options) {
	element.addEventListener(name, handler, options)
	signal.addEventListener('abort', () => {	
		element.removeEventListener(name, handler, options)
	})
}
```
😬

Reply:
> This however is one of those situations where listeners will just pile up in Safari

Aren't listeners natively deduplicated if they're used with named functions? Unless Safari doesn't do it

Reply:
Yes but:

- we’re  passing a new signal every time, which should break the memoization
- I wonder if Safari ignores the signal in the memoization since it doesn’t support it
- This is just one situation, let’s see if there are other `onetime` usages that use other observers.